packages/integration-karma/test/light-dom/assigned-node-element/index.spec.js

```diff
@@ -0,0 +1,110 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import { extractDataIds } from 'test-utils';
+
+import LightContainer from 'x/lightContainer';
+import ShadowContainer from 'x/shadowContainer';
+
+describe('#2386 - Light DOM component slotted content in shadow slots', () => {
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+    });
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+
+    describe('light -> shadow', () => {
+        let elm, nodes;
+        beforeEach(() => {
+            elm = createElement('x-light-container', {
+                is: LightContainer,
+            });
+            document.body.appendChild(elm);
+            nodes = extractDataIds(elm);
+        });
+
+        it('assignedNodes', () => {
+            expect(nodes.slot.assignedNodes()).toEqual([nodes.p, nodes.p.nextSibling]);
+        });
+        it('assignedElements', () => {
+            expect(nodes.slot.assignedElements()).toEqual([nodes.p]);
+        });
+        it('assignedSlot', () => {
+            expect(nodes.p.assignedSlot).toEqual(nodes.slot);
+        });
+        it('childNodes', () => {
+            expect(Array.from(nodes.slot.childNodes)).toEqual([]);
+        });
+        it('parentNode', () => {
+            expect(nodes.p.parentNode).toEqual(nodes.consumer);
+        });
+        it('parentElement', () => {
+            expect(nodes.p.parentElement).toEqual(nodes.consumer);
+        });
+        xit('getRootNode', () => {
+            expect(nodes.p.getRootNode()).toEqual(document);
```

Comment:
It's a little more complex than this. `Node.prototype.getRootNode` fails when a light DOM element is the top-level light DOM element because there is no owner key to inherit from.